contracts/voting/dao-voting-cw4/src/contract.rs

```diff
@@ -24,124 +25,79 @@ pub fn instantiate(
-    for diff in diffs {
-        let user_address = deps.api.addr_validate(&diff.key)?;
-        let weight = diff.new.unwrap_or_default();
-        let old = diff.old.unwrap_or_default();
-        // Do we need to add to positive difference or negative difference
-        if weight > old {
-            positive_difference += Uint128::from(weight - old);
-        } else {
-            negative_difference += Uint128::from(old - weight);
+            // We need to set ourself as the CW4 admin it is then transferred to the DAO in the reply
+            let msg = WasmMsg::Instantiate {
+                admin: Some(info.sender.to_string()),
+                code_id: cw4_group_code_id,
+                msg: to_binary(&cw4_group::msg::InstantiateMsg {
+                    admin: Some(env.contract.address.to_string()),
+                    members: initial_members,
+                })?,
+                funds: vec![],
+                label: env.contract.address.to_string(),
+            };
+
+            let msg = SubMsg::reply_on_success(msg, INSTANTIATE_GROUP_REPLY_ID);
+
+            Ok(Response::new()
+                .add_attribute("action", "instantiate")
+                .add_submessage(msg))
         }
+        GroupContract::Existing { address } => {
+            let group_contract = deps.api.addr_validate(&address)?;
+            GROUP_CONTRACT.save(deps.storage, &group_contract)?;
 
-        if weight != 0 {
-            USER_WEIGHTS.save(
-                deps.storage,
-                &user_address,
-                &Uint128::from(weight),
-                env.block.height,
-            )?;
-        } else if weight == 0 && weight != old {
-            // This works because query_voting_power_at_height will return 0 on address missing
-            // from storage, so no need to store anything.
-            //
-            // Note that we also check for weight != old: If for some reason this hook is triggered
-            // with weight 0 for old and new values, we don't need to do anything.
-            USER_WEIGHTS.remove(deps.storage, &user_address, env.block.height)?;
+            Ok(Response::new()
+                .add_attribute("action", "instantiate")
+                .add_attribute("group_contract", "address"))
         }
     }
-    let new_total_weight = total_weight
-        .checked_add(positive_difference)
-        .map_err(StdError::overflow)?
-        .checked_sub(negative_difference)
-        .map_err(StdError::overflow)?;
-    TOTAL_WEIGHT.save(deps.storage, &new_total_weight, env.block.height)?;
-
-    Ok(Response::new()
-        .add_attribute("action", "member_changed_hook")
-        .add_attribute("total_weight", new_total_weight.to_string()))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    _deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    _msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    Err(ContractError::NoExecute {})
```

Comment:
Complains no tests, I guess you can just add a simple test that trying to execute something errors? 
should be a good reminder in the future to add test if execute msgs are added as well